spring-xd-messagebus-kafka/src/main/java/org/springframework/xd/dirt/integration/kafka/KafkaMessageBus.java

```diff
@@ -377,20 +384,50 @@ public void bindReplier(String name, MessageChannel requests, MessageChannel rep
 	 * Creates a Kafka topic if needed, or try to increase its partition count to the desired number.
 	 */
 	private void ensureTopicCreated(final String topicName, int numPartitions, int replicationFactor) {
+
 		final int sessionTimeoutMs = 10000;
 		final int connectionTimeoutMs = 10000;
-		ZkClient zkClient = new ZkClient(zkAddress, sessionTimeoutMs, connectionTimeoutMs, utf8Serializer);
-
-		// The following is basically copy/paste from AdminUtils.createTopic() with
-		// createOrUpdateTopicPartitionAssignmentPathInZK(..., update=true)
-		Properties topicConfig = new Properties();
-		Seq<Object> brokerList = ZkUtils.getSortedBrokerList(zkClient);
-		scala.collection.Map<Object, Seq<Object>> replicaAssignment = AdminUtils.assignReplicasToBrokers(brokerList,
-				numPartitions, replicationFactor, -1, -1);
-		AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK(zkClient, topicName, replicaAssignment, topicConfig,
-				true);
-		zkClient.close();
+		final ZkClient zkClient = new ZkClient(zkAddress, sessionTimeoutMs, connectionTimeoutMs, utf8Serializer);
+		try {
+			// The following is basically copy/paste from AdminUtils.createTopic() with
+			// createOrUpdateTopicPartitionAssignmentPathInZK(..., update=true)
+			Properties topicConfig = new Properties();
+			Seq<Object> brokerList = ZkUtils.getSortedBrokerList(zkClient);
+			scala.collection.Map<Object, Seq<Object>> replicaAssignment = AdminUtils.assignReplicasToBrokers(brokerList,
+					numPartitions, replicationFactor, -1, -1);
+			AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK(zkClient, topicName, replicaAssignment, topicConfig,
+					true);
+
+			RetryTemplate retryTemplate = new RetryTemplate();
+
+			CompositeRetryPolicy policy = new CompositeRetryPolicy();
+			TimeoutRetryPolicy timeoutRetryPolicy = new TimeoutRetryPolicy();
+			timeoutRetryPolicy.setTimeout(METADATA_VERIFICATION_TIMEOUT);
+			SimpleRetryPolicy simpleRetryPolicy = new SimpleRetryPolicy();
+			simpleRetryPolicy.setMaxAttempts(METADATA_VERIFICATION_RETRY_ATTEMPTS);
+			policy.setPolicies(new RetryPolicy[] {timeoutRetryPolicy, simpleRetryPolicy});
 
+			try {
+				retryTemplate.execute(new RetryCallback<Boolean, Exception>() {
+					@Override
+					public Boolean doWithRetry(RetryContext context) throws Exception {
+						TopicMetadata topicMetadata = AdminUtils.fetchTopicMetadataFromZk(topicName, zkClient);
```

Comment:
I think for such an low-level resource we need some `BackOffPolicy` between each retry. By default it is `NoBackOffPolicy`. 
The `TimeoutRetryPolicy` just says how long our `retryTemplate.execute()` can live, but it doesn't help to have some `Thread.sleep` between retries.
Am I missing something here otherwise ?